src/main/java/roomescape/controller/ReservationController.java

```diff
@@ -0,0 +1,78 @@
+package roomescape.controller;
+
+import java.net.URI;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicLong;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.ResponseBody;
+import roomescape.dto.ReservationRequestDto;
+import roomescape.dto.ReservationResponseDto;
+import roomescape.mapper.DtoMapper;
+import roomescape.model.Reservation;
+import roomescape.utils.RequestValidator;
+
+@Controller
+public class ReservationController {
+
+    private List<Reservation> reservations;
+    private final AtomicLong idMaker = new AtomicLong();
+
+    public ReservationController() {
+        this.reservations = new ArrayList<>();
+    }
+
+    @GetMapping("/reservation")
+    public String getReservation() {
+        return "reservation";
+    }
+
+    @GetMapping("/reservations")
+    @ResponseBody
+    public List<ReservationResponseDto> getReservations() {
+        return reservations.stream()
+                .map(DtoMapper::convertToDTO)
+                .toList();
+    }
+
+    @PostMapping("/reservations")
+    public ResponseEntity<ReservationResponseDto> saveReservation(@RequestBody ReservationRequestDto dto) {
+        RequestValidator.validateSaveRequest(dto);
+        final Reservation reservation = new Reservation(
+                idMaker.incrementAndGet(),
+                dto.getName(),
+                dto.getDate(),
+                dto.getTime());
+        reservations.add(reservation);
+        final ReservationResponseDto responseDto = DtoMapper.convertToDTO(reservation);
+        return ResponseEntity
+                .created(URI.create("/reservations/" + responseDto.getId()))
+                .body(responseDto);

```

Comment:
```suggestion
        return ResponseEntity.status(HttpStatus.CREATED).body(responseDto);
```
응답 Body에 생성된 ReservationDto 객체를 넣어주는 것이 좋아보여요!

Reply:
`POST /reservations` 요청의 경우, JSON 응답과 함께 `Location` 응답 헤더에 `/reservations/{생성된 ID}` 값을 넣어 반환해야 하는 걸로 알고 있어요! 제시해주신 코드를 사용했을 땐 API 응답에 `Location` 필드 값이 정의되지 않아 삼단계 테스트 코드를 실패하는 걸로 나오는데 제시해주신 코드의 의도를 알 수 있을까요?